### DISCUSSION TITLE
How can I tell Renovate to only update the PR when the base branch has changed, not when a new version is available ?

### DISCUSSION THREAD

**Discussion by victordelajarte** [renovatebot/renovate#26294] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

I have a renovate PR to update a dependency, it's nice to have it rebased if the target branch has changed, but here it changes when the new dependency version is accepted by my configuration.

Ideally, Renovate would open the PR with a specific version, and the publication of a new version should not update the PR, just the Dependency Dashboard. 
But if I push an update to the base branch, it should rebase the PR automatically (cf [rebaseWhen option](https://docs.renovatebot.com/configuration-options/#rebasewhen))

I read the documentation but couldn't find the configuration option for this behavior, I don't know if it is possible. 
If not, I will set `rebaseWhen` to `never` but I'm not even sure that it will work, and I'll lose the auto-rebasing feature.

```
// config
{
      "matchManagers": ["npm"],
      "minimumReleaseAge": "10 days",
      "internalChecksFilter": "strict"
}
```

<img width="868" alt="Capture d’écran 2023-12-14 à 11 51 59" src="https://github.com/renovatebot/renovate/assets/29466815/966e5733-9a97-4819-a040-e3f0643c86d9">

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Please clarify:
1. You want the PR rebased if there's a commit to the base branch?
2. You never want the versions in a PR to be changed once the PR is updated?

**Reply by victordelajarte** [reply]
Yes, ideally I want both. If the PR is created for version X, it will always be for version X (1.115.4 in the screenshot), even if new versions are published. 
And if there are changes on the base branch, a rebase is nice so I don't have to trigger it manually to check the CI before merging

**Reply by rarkins** [reply, marked as answer]
Unfortunately it's not easily achievable to do both. This is because Renovate does not "actually" rebase using git, instead it completely reapplies the updates to a branch on top of the latest commit of the base branch. i.e. it's reapplying it's changes, not rebasing its branch, although the two of those seem similar in final result.

When Renovate applies changes, it applies whatever is the latest config, and right now it does not have a way to say to itself "ignore whatever you think should be there, instead match exactly what's already in the branch".

**Reply by victordelajarte** [reply]
OK, interesting, thanks ! So `rebaseWhen: 'never'` should do the trick to avoid the rebase ? 
But if I trigger the rebase manually (with the checkbox), the new version will be used so I need to do a real rebase with git ?

**Reply by rarkins** [reply]
Yes to both
